test/unit/producer.test.ts

```diff
@@ -0,0 +1,58 @@
+import { expect } from "chai"
+import { randomUUID } from "crypto"
+import { connect } from "../../src"
+import { Rabbit } from "../support/rabbit"
+
+describe("Producer", () => {
+  const rabbit = new Rabbit()
+  const testStreamName = "test-stream"
+  const publisherRef = randomUUID()
+
+  beforeEach(async () => {
+    await rabbit.createStream(testStreamName)
+  })
+
+  afterEach(async () => {
+    await rabbit.deleteStream(testStreamName)
+  })
+
+  describe("Send: ", () => {
+    it("boot is true", async () => {
```

Comment:
if try to read the test as sentence this name are not very explicative. I suggest a tet like:
```suggestion
    it("retrive publishing id from server when boot is true", async () => {
```

and in the test i suggest to firs send some message using one publishRef, and then using the same publishRef see that we publish with new id ...

same for boot false, we overwrite the stream.